tests/components/websocket_api/test_http.py

```diff
@@ -392,7 +392,7 @@ async def test_prepare_fail_connection_reset(
             "homeassistant.components.websocket_api.http.web.WebSocketResponse.prepare",
             side_effect=(ConnectionResetError, web.WebSocketResponse.prepare),
         ),
-        pytest.raises(WSServerHandshakeError),
+        pytest.raises(ServerDisconnectedError),
```

Comment:
We are patching out `prepare`, `aiohttp` now notices this and it fails sooner so the exception has changed on the client side.  This is expected